src/org/alicebot/ab/ProgramABListener.java

```diff
@@ -11,20 +11,20 @@
    * @param predicateName
    * @param result
    */
-  public void onChangePredicate(String predicateName, String result);
+  public void onChangePredicate(Chat chat, String predicateName, String result);
 
   /**
    * Callback when category is added from a the result of matching a learnf tag
    * 
    * @param c - the category added
    */
-  public void onLearnF(Category c);
+  public void onLearnF(Bot bot, Category c);
 
   /**
    * Callback when category is added from a the result of matching a learn tag
    * 
    * @param c - the category added
    */
-  public void onLearn(Category c);
+  public void onLearn(Bot bot, Category c);
```

Comment:
This should be specific to a chat session so you know what session it came from so you know what username to associate with the learn tag.  (similar to the learnf tag, it'd be useful to know which chat session it's associated with.)